tests/chainerx_tests/unit_tests/routines_tests/test_math.py

```diff
@@ -282,6 +283,52 @@ def _make_same_in_out_dtypes(number_of_in_params, dtypes):
     _in_out_dtypes_float_array_float_scalar)
 
 
+def _permutate_shapes(shapes_list):
+    # Permutates input shapes
+    permutated_shapes_list = []
+    for in_shape1, in_shape2 in shapes_list:
+        permutated_shapes_list.append((in_shape1, in_shape2))
+        permutated_shapes_list.append((in_shape2, in_shape1))
+    return list(set(permutated_shapes_list))
+
+
+_shapes_combination_inplace_binary = [
+    # Same shapes
+    ((1,), (1,)),
+    ((3, 4), (3, 4)),
+    # Broadcast
+    ((10,), (1,)),
+    ((3, 4), (3, 1)),
+    ((3, 4), (1, 4)),
+    ((3, 4), (4,)),
+    ((3, 4), (1, 1)),
+    ((3, 4), (1,)),
+    ((2, 3, 4), (1, 1, 1)),
+    # 0-dim shape
+    ((), ()),
+    ((1,), ()),
+    ((3,), ()),
+    ((2, 3), ()),
+    # 0-size shape
+    ((0,), (0,)),
+    ((0,), (1,)),
+    ((0,), ()),
+    ((2, 0, 3), (2, 0, 3)),
+    # TODO(imanishi): Fix strides
+    # ((2, 0, 3), (0, 1)),
```

Comment:
```
E       chainer.testing.function_link.FunctionTestError: Strides do not match with NumPy outputs.
E       Expected shapes and dtypes: (2, 0, 3):int8
E       Actual shapes and dtypes:   (2, 0, 3):int8
E       Expected strides: (0, 3, 1)
E       Actual strides:   (3, 3, 1)
```

Reply:
For the record, e.g. `Add` and please note that not all dtypes are tested.